From f91ef9328c48ba9dfb5b81a2c46f8c48b25fdaa1 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Fri, 23 Jun 2017 18:22:46 +0000
Subject: [PATCH] DOCKER-1072 Work around OS-6193 in tools/check-docs.sh
 Reviewed by: Dave Eddy <dave@daveeddy.com> Reviewed by: Trent Mick
 <trent.mick@joyent.com> Approved by: Trent Mick <trent.mick@joyent.com>

---
 deps/javascriptlint |  2 +-
 tools/check-docs.sh | 19 ++++++++++++-------
 2 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/deps/javascriptlint b/deps/javascriptlint
index e1bd0ab..ad52812 160000
--- a/deps/javascriptlint
+++ b/deps/javascriptlint
@@ -1 +1 @@
-Subproject commit e1bd0abfd424811af469d1ece3af131d95443924
+Subproject commit ad52812e77bdfb1e90fb71a1201adb2b665a27e6
diff --git a/tools/check-docs.sh b/tools/check-docs.sh
index cbba338..854f864 100755
--- a/tools/check-docs.sh
+++ b/tools/check-docs.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 #
@@ -29,18 +29,23 @@ fi
 set -o errexit
 set -o pipefail
 
-TOP=$(cd $(dirname $0)/../; pwd)
+TOP=$(cd "$(dirname "$0")/../"; pwd)
 
 
 #---- mainline
 
 hits=0
-for file in $(find $TOP/docs -name "*.md"); do
-    lastchar=$(tail -c1 $file | od -a | head -1 | awk '{print $2}')
-    if [[ $lastchar != 'nl' ]]; then
+
+# Check each of the files in docs/. (We use -print0 in case files
+# have spaces in their name.)
+while read -d $'\0' -r file; do
+    # Map the newline character into a space, and spaces into
+    # an underscore, since bash likes to trim trailing newlines
+    # from command outputs.
+    if [[ `tail -1c "$file" | tr ' \n' '_ '` != ' ' ]]; then
         echo "$file: does not end with a newline" >&2
-        hits=$(( $hits + 1 ))
+        hits=1
     fi
-done
+done < <(find "$TOP/docs" -name "*.md" -print0)
 
 exit $hits
-- 
2.21.0

